src/main/java/com/gotocompany/firehose/evaluator/GrpcResponseCelPayloadEvaluator.java

```diff
@@ -0,0 +1,60 @@
+package com.gotocompany.firehose.evaluator;
+
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.Message;
+import com.gotocompany.firehose.utils.CelUtils;
+import dev.cel.common.types.CelKind;
+import dev.cel.compiler.CelCompiler;
+import dev.cel.runtime.CelRuntime;
+import dev.cel.runtime.CelRuntimeFactory;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Implementation of PayloadEvaluator that evaluates gRPC responses using CEL (Common Expression Language).
+ */
+@Slf4j
+public class GrpcResponseCelPayloadEvaluator implements PayloadEvaluator<Message> {
+
+    private final Descriptors.Descriptor descriptor;
+    private CelRuntime.Program celProgram;
+
+    /**
+     * Constructs a GrpcResponseCelPayloadEvaluator with the specified descriptor and CEL expression.
+     *
+     * @param descriptor    the descriptor of the gRPC message
+     * @param celExpression the CEL expression to evaluate against the message
+     */
+    public GrpcResponseCelPayloadEvaluator(Descriptors.Descriptor descriptor, String celExpression) {
+        this.descriptor = descriptor;
+        buildCelEnvironment(celExpression);
+    }
+
+    /**
+     * Evaluates the given gRPC message payload using the CEL program.
+     *
+     * @param payload the gRPC message to be evaluated
+     * @return true if the payload passes the evaluation, false otherwise
+     */
+    @Override
+    public boolean evaluate(Message payload) {
+        if (!descriptor.getFullName().equals(payload.getDescriptorForType().getFullName())) {
+            throw new IllegalArgumentException("Payload does not match descriptor");
```

Comment:
throw new IllegalArgumentException(fomatString - "Payload %s does not match descriptor %s" - parameters);